crates/bevy_ecs/src/query/state.rs

```diff
@@ -95,9 +99,22 @@ impl<Q: WorldQuery, F: ReadOnlyWorldQuery> QueryState<Q, F> {
 
 impl<Q: WorldQuery, F: ReadOnlyWorldQuery> QueryState<Q, F> {
     /// Creates a new [`QueryState`] from a given [`World`] and inherits the result of `world.id()`.
-    pub fn new(world: &mut World) -> Self {
-        let fetch_state = Q::init_state(world);
-        let filter_state = F::init_state(world);
+    pub fn new(world: &mut World) -> Self
+    where
+        <Q as WorldQuery>::Config: Default,
+        <F as WorldQuery>::Config: Default,
+    {
+        QueryState::new_with_config(world, Default::default(), Default::default())
```

Comment:
```suggestion
        let fetch_config = Default::default();
        let filter_config = Default::default();
        QueryState::new_with_config(world, fetch_config, filter_config)
```

Cosmetic, but I think this makes the code significantly easier to follow at a glance.